### PR TITLE
#352 update data parsing depth

### LINKED ISSUE #352: Update data parsing to traverse to pack size depth
Data parsing logic is skipping entities.

## Expected Behaviour

Data parsing logic should include `Unit` and parse to depth of `PackSize`.

## Current Behaviour

Current data parsing logic stops at the `DoseStrength` (incorrectly labelled as `Unit`).

## Possible Solution

Update `DataParser` and `DataLoader` logic.

### PR DESCRIPTION
**Pull request by wlthomson**
Fixes #352.

Branched off #351.

## Description

Extends data parser to UC9 (pack size).

### Documentation:

Tick below if one of the following applies:

1. Documentation has been added to cover changes introduced by this PR.
2. This PR requires no changes to documentation.

- [x] Documentation OK.

### Tests:

Tick below if one of the following applies:

1. Tests have been added to cover changes introduced by this PR.
2. This PR requires no new tests.

- [x] Tests OK.

### CODE REVIEW

File: unified-codes/apps/data-service/src/app/v1/types/DgraphDataSource.ts [hunk 1]
```diff
@@ -35,13 +43,13 @@ export class DgraphDataSource extends RESTDataSource {
       switch (type) {
         case EEntityType.FORM_CATEGORY:
         case EEntityType.FORM:
-          return 'Form';
+          return EEntityTypeV2.Form;
```
Comment (katherine-sussol) [inline unified-codes/apps/data-service/src/app/v1/types/DgraphDataSource.ts:46]: 🥳

File: unified-codes/apps/data-service/src/app/v1/types/DgraphDataSource.ts [hunk 2]
```diff
@@ -333,34 +339,77 @@ export class DgraphDataSource extends RESTDataSource {
               route.children.forEach(doseForm => {
                 if (doseForm.children) {
                   doseForm.children.forEach(doseQualifierOrDoseStrength => {
-                      if (doseQualifierOrDoseStrength.children) {
-                        doseQualifierOrDoseStrength.children.forEach(doseStrength => {
+                      if (doseQualifierOrDoseStrength.type === EEntityTypeV2.DoseStrength) {
```
Comment (katherine-sussol) [inline unified-codes/apps/data-service/src/app/v1/types/DgraphDataSource.ts:342]: This works well so I think all good for now! May be nice later to banish the mSupply specific stuff to its own file 🧙

File: unified-codes/libs/data/src/lib/v1/EntitySort.ts [hunk 3]
```diff
@@ -1,5 +1,10 @@
 import { EEntityField } from './Entity';
 
+export enum EEntitySortOrder {
+  Asc = 'orderasc',
```
Comment (katherine-sussol) [inline unified-codes/libs/data/src/lib/v1/EntitySort.ts:4]: Oh this is where that missing enum from #350 was hiding! 👍

File: unified-codes/libs/data/src/lib/v2/index.ts [hunk 4]
```diff
@@ -0,0 +1,2 @@
+export * from './entity';
```
Comment (katherine-sussol) [inline unified-codes/libs/data/src/lib/v2/index.ts:1]: I'm still getting some warnings here due to casing, still worth capitalising?

File: unified-codes/libs/data/src/lib/v2/property/DrugInteractionProperty.ts [hunk 5]
```diff
@@ -42,4 +42,4 @@ export class DrugInteractionProperty extends Property implements IDrugInteractio
   }
 }
 
-export default DrugInteraction;
+export default DrugInteractionProperty;
```
Comment (katherine-sussol) [inline unified-codes/libs/data/src/lib/v2/property/DrugInteractionProperty.ts:45]: 🥳

File: unified-codes/tools/scripts/dgraph/data/v2/products.csv [hunk 6]
```diff
@@ -1,1185 +1,1185 @@
-﻿Product ,Product Synonym,Combination,Route,Dose Form,Dose Qualification,Strength,Unit of Presentation,Immediate Packaging,Pack Size,Outer Packaging,Brand,UC1,"UC2 
+Product ,Product Synonym,Combination,Route,Dose Form,Dose Qualification,Strength,Unit of Presentation,Immediate Packaging,Pack Size,Outer Packaging,Manufacturer,Brand,UC1,"UC2
```
Comment (katherine-sussol) [inline unified-codes/tools/scripts/dgraph/data/v2/products.csv:1]: Uhh...`@@ -1,1185 +1,1185 @@ `??

LGTM!

**Review by katherine-sussol — APPROVED** [review]
Nice! LGTM.

I don't know what's going on with the CI, it's having issues finding properties but it's working on my machine? 🤷

File: unified-codes/apps/data-service/src/app/v1/types/DgraphDataSource.ts [hunk 7]
```diff
@@ -50,12 +58,11 @@ export class DgraphDataSource extends RESTDataSource {
...
         case EEntityCategory.DRUG:
-          return 'Drug';
+          return EEntityCategoryV2.DRUG;
         case EEntityCategory.MEDICINAL_PRODUCT:
-          return 'Consumable';
+          return EEntityCategoryV2.CONSUMABLE;
         case EEntityCategory.OTHER:
-          return 'Other';
-      }
+          return EEntityCategoryV2.OTHER;      }
```
[the remark this answers is not in the document]
Comment (wlthomson) [reply unified-codes/apps/data-service/src/app/v1/types/DgraphDataSource.ts]: 37b626ae8.

**Pull request merged**